File: checker_test/ac/submission.dart
import 'dart:io';
import 'dart:collection';

void main() {
    int? n = int.tryParse(stdin.readLineSync() ?? '');

    // take list input
    List<int?> arr =  stdin.readLineSync()?.split(' ').map((e) => int.parse(e)).toList() ?? [];

    int? maxx = arr.reduce((value, element) => value! > element! ? value : element);

    for (int i = 0; i < n!; i++) {
        if (arr[i] == maxx) {
            print(i);
            break;
        }
    }
}

File: checker_test/ac/submission.pl
use strict;
use warnings;

my $n = <STDIN>;
chomp $n;

my $line = <STDIN>;
chomp $line;

my @arr = split(' ', $line);

my $max_index = -1;
my $max_value = -1;

for (my $i = 0; $i < $n; $i++) {
    if ($arr[$i] > $max_value) {
        $max_index = $i;
        $max_value = $arr[$i];
    }
}

print $max_index, "\n";